src/pretix/control/templatetags/build_locale_url.py

```diff
@@ -0,0 +1,21 @@
+import urllib.parse
+from django import template
+
+register = template.Library()
+
+
+@register.filter
+def build_locale_url(request, locale_code):
+    """
+    Constructs the locale change URL by appending the locale and
+    next path and query string to the URL.
+    """
+    # Build base URL with locale
+    base_url = f"{request.path}?locale={locale_code}"
+
+    # Append query string if it exists
+    if request.META.get('QUERY_STRING'):
+        query_string = urllib.parse.urlencode(urllib.parse.parse_qsl(request.META['QUERY_STRING']))
+        return f"{base_url}%3F{query_string}"
```

Comment:
The point of using `urllib` is to let it automatically encode the illegal characters and construct query string in safe way, why do you duplicate its functionality by writing "?locale=" and "%3F"?

And why do you do:

```py
urlencode(parse_qsl(request.META['QUERY_STRING']))
```

which is the same as accessing `request.META['QUERY_STRING']` ?

Reply:
Hi @hongquan, thanks for your review, I have updated the code,
using parse_qs to parse the existing query string and add its parameters to the new params then construct the full URL-encoded query string